Add org.jitsi.impl.neomedia.RawPacket.DISCARD_CONTRIBUTING_SOURCES config property

debugmaster opened the pull request:
This property is useful when a SIP Provider does not support the list of contributing sources in its SIP infrastructure. If one is having annoying noise issues, this property may solve it when it is set to true.

jitsi-jenkins commented:
Hi, thanks for your contribution!
If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

debugmaster commented:
I've signed the CLA

bgrozev commented:
Thank you for the contribution. However, I don't think the basic RawPacket class is the right place for such logic. It should go to MediaStreamImpl or CsrcTransformEngine, or whatever code you are using which calls setCsrcList()

debugmaster commented:
I reverted all changes from RawPakcet and moved to CsrcTranformEngine, which is the only class that calls setCsrcList().

I added the property verification to [this line](https://github.com/jitsi/libjitsi/pull/242/files/f8ea91cf1c5b8d200d9c4ccf053364e7d3e5a850..66bd7c3cc46e55de03e7f1d4355d057998f11183#diff-f1d6d3bb587859167d31a6f97ca08d82R313). Tell me if I should place it somewhere else.

bgrozev commented:
Jenkins, test this please

bgrozev commented:
@debugmaster LGTM
